packages/flutter_tools/lib/src/commands/devices.dart

```diff
@@ -60,34 +60,39 @@ class DevicesCommand extends FlutterCommand {
 
     if (boolArg('machine')) {
       await printDevicesAsJson(devices);
-    } else if (devices.isEmpty) {
-      final StringBuffer status = StringBuffer('No devices detected.');
-      status.writeln();
-      status.writeln();
-      status.writeln('Run "flutter emulators" to list and start any available device emulators.');
-      status.writeln();
-      status.write('If you expected your device to be detected, please run "flutter doctor" to diagnose potential issues. ');
-      if (timeout == null) {
-        status.write('You may also try increasing the time to wait for connected devices with the --timeout flag. ');
-      }
-      status.write('Visit https://flutter.dev/setup/ for troubleshooting tips.');
-
-      globals.printStatus(status.toString());
-      final List<String> diagnostics = await deviceManager.getDeviceDiagnostics();
-      if (diagnostics.isNotEmpty) {
-        globals.printStatus('');
-        for (final String diagnostic in diagnostics) {
-          globals.printStatus('• $diagnostic', hangingIndent: 2);
+    } else {
```

Comment:
Tip: this one is easier to review with whitespace changes ignored:
![Screen Shot 2020-05-18 at 3 20 17 PM](https://user-images.githubusercontent.com/682784/82265117-1d1fa000-991b-11ea-9ee9-6ff3a803d0c0.png)